### DISCUSSION TITLE
Analog button value range

### DISCUSSION THREAD

**Discussion by karim622** [rlogiacco/AnalogButtons#14] [Q&A] [answered]
Greetings thanks for the great work,
may I know if its possible to use a range of values instead of a specific button value?

Button b1= Button(170, &b1Click, &b1Hold);

instead of using 170 as a specific value, I want to use a range i.e. 165~175

Thank you

**Comment by rlogiacco** [comment, marked as answer]
Actually _every_ value should be considered as a range:

> the `analog value margin` which takes into account slight resistance fluctuations and ADC errors transforming the button value into a range (defaults to 10)

That's from the [Analog pin definition](https://github.com/rlogiacco/AnalogButtons#2-analog-pin-definition) documentation.

```cpp
AnalogButtons(uint8_t pin, uint8_t mode, uint16_t debounce, uint8_t margin)
```

So, if your request is to have a margin customizable for each button that will require additional memory (which I always strive to keep as low as possible). The library can easily be adapted for such corner-case, but is it worth it?
I will accept a pull request to cover that, but only if it can be switched off via a macro so that only those needing it will pay the extra memory cost.

**Comment by rlogiacco** [comment]
BTW, your `170` is actually `[160-180]` (boundaries included) due to the default margin value of `10`
